r-package/grf/vignettes/rate.Rmd

```diff
@@ -0,0 +1,216 @@
+
+## Quantifying treatment benefit: the Targeting Operator Characteristic
+As outlined in the previous section, we are interested in the presence of heterogeneity and how much benefit there is to prioritizing treatment based on this heterogeneity using a targeting rule $S(X_i)$. By "benefit" we mean the expected increase in outcomes from giving treatment to a fraction of the population with the largest scores $S(X_i)$ as opposed to giving treatment to a randomly selected fraction of the same size.
+
+As a first step towards a suitable metric that helps us asses this, some visual aid would be nice. One reasonable thing to do would be to chop the population up into groups defined by $S(X_i)$, then compare the ATE in these groups to the overall ATE from treating everyone, then plot this over all groups. This is what the *Targeting Operator Characteristic* (TOC) does, where each group is the top q-th fraction of individuals with the largest prioritization score $S(X_i)$.
+
+Let $F$ be the distribution function of $S(X_i)$ and $q \in (0, 1]$ the fraction of samples treated, the TOC at $q$ is then defined by
+
+\begin{equation*}
+\begin{split}
+&\textrm{TOC}(q) = E[Y_i(1) - Y_i(0) \,|\, S(X_i) \geq F^{-1}_{S(X_i)}(1 - q)]\\
+& \ \ \ \ \ \ \ \ \ \ \ \ \ \ \ \  - E[Y_i(1) - Y_i(0)],
+\end{split}
+\end{equation*}
+
+where $F_{S(X_i)}$ is the distribution function of $S(X_i)$.The TOC curve is motivated by the *Receiver Operating Characteristic* (ROC), a widely used metric for assessing the performance of a classification rule.
+
+As a toy example, suppose we let $\tau(X_i) = X_1$ where $X_1 \sim N(0, 1)$. Then the TOC curve when prioritizing using $S(X_i) = \tau(X_i)$ looks like:
+
+```{r toc-plot, echo=FALSE}
+n <- 1000
+p <- 1
+X <- matrix(rnorm(n * p), n, p)
+tau <- X[, 1]
+
+ATE <- mean(tau)
+sort.idx <- order(tau, decreasing = TRUE)
+TOC <- rep(NA, n)
+for (i in 1:n) {
+  TOC[i] <- mean(tau[sort.idx[1:i]]) - ATE
+}
+q <- seq(1/n, 1, by = 1/n)
+df <- data.frame(q, TOC)
+ypoint <- df[df$q == 0.05, "TOC"]
+
+ggplot(df, aes(x = q, y = TOC)) +
+  geom_line() +
+  geom_hline(yintercept = 0, lty = 3) +
+  annotate(geom = 'text', x = .1, y = ypoint, color = 'black', hjust = -0.1,
+           label = 'ATE of top 5% minus ATE') +
+  annotate("segment", x = .1, xend = .05, y = ypoint, yend = ypoint,
+           arrow = arrow(type = "closed", length = unit(0.02, "npc"))) +
+  theme_classic() +
+  labs(y = "", title = "Targeting Operator Characteristic")
+```
+
+Note how high the TOC is for the first few units treated: this is intuitive since our toy CATEs are normally distributed with mean zero, if we identify and treat only the top 5% of units with treatment effects greater than $z_{0.95}$, the benefit over treating everyone (0) is large. As we move further along, mixing in more people with lower treatment effect, we get closer to the ATE, until at $q=1$ we equal it.
```

Comment:
"Note how high the TOC is for the first few units treated**.** This is intuitive since our toy CATEs are normally distributed with mean zero**;** if we identify and treat only the top 5% of units with treatment effects greater than $z_{0.95}$, the benefit over treating everyone (0) is large."